blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -81,14 +81,17 @@ public static void send(Message message, OutputStream outputStream) throws IOExc
             toSend.append("\n");
         }
         toSend.append("\n");
-        outputStream.write(toSend.toString().getBytes());
+        synchronized (outputStream) {
+            outputStream.write(toSend.toString().getBytes());
```

Comment:
```
socket.getChannel().write(toSend); 
```
This throws the Null Exception and I think it's causes by we just create the socket in blocking mode. see StompMangement.connect()

```
Socket socket = new Socket(host, port);
InputStream inputStream = socket.getInputStream();
OutputStream outputStream = socket.getOutputStream();
```
I'm trying to do 

```
SocketChannel channel=SocketChannel.open();
channel.connect(new InetSocketAddress(host,port));
channel.finishConnect();

Socket socket = channel.socket();
```
It looks good but the SompManagement.receive() now is blocking on reading from the input stream. I think it could also need to change the receive method by using the nio channel and this would require more code changes. So I agree assemble the data and synchronize the write operation.